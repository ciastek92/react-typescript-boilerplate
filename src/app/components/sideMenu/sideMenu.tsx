import { Hamburger } from 'app/components/hamburger/hamburger';
import SideNav from 'app/components/nav/sideNav';
import cx from 'classnames';
import React from 'react';

interface State {
  isMenuOpened: boolean;
}

export class SideMenu extends React.Component<{}, State> {
  public state: State = {
    isMenuOpened: false
  };

  public menuToggle = () => {
    const { isMenuOpened } = this.state;
    this.setState({
      isMenuOpened: !isMenuOpened
    });
  };

  public render() {
    const { isMenuOpened } = this.state;
    const headerContainerClassName = cx(
      'o-aside-layout__burger',
      'u-display-md--none',
      'u-cursor--pointer',
      'js-hit-area',
      { 'u-display--none': isMenuOpened }
    );
    const sideBurgerContainerClassName = cx(
      'c-side-nav__burger',
      'u-cursor--pointer',
      'js-hit-area',
      { 'u-display--none': !isMenuOpened }
    );
    const hamburger = (
      <Hamburger open={isMenuOpened} darkTheme={!isMenuOpened} />
    );
    return (
      <React.Fragment>
        <div className={headerContainerClassName} onClick={this.menuToggle}>
          {hamburger}
        </div>
        <aside
          className={cx('o-aside-layout__sidebar', {
            'o-aside-layout__sidebar--show': isMenuOpened
          })}
        >
          <SideNav>
            <div
              className={sideBurgerContainerClassName}
              onClick={this.menuToggle}
            >
              {hamburger}
            </div>
          </SideNav>
        </aside>
      </React.Fragment>
    );
  }
}
