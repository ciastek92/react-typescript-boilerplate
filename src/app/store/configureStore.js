import { applyMiddleware, compose, createStore } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import { rootReducer } from './store.reducer';

export default function configureStore(history, initialState, epicMiddleware) {
  const middleware = [epicMiddleware, routerMiddleware(history)];

  // In development, use the browser's Redux dev tools extension if installed
  const enhancers = [];
  const isDevelopment = process.env.NODE_ENV === 'development';
  if (
    isDevelopment &&
    typeof window !== 'undefined' &&
    window.__REDUX_DEVTOOLS_EXTENSION__
  ) {
    enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION__());
  }

  return createStore(
    rootReducer,
    initialState,
    compose(
      applyMiddleware(...middleware),
      ...enhancers
    )
  );
}
