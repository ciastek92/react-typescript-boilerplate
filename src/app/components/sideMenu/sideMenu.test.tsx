import { shallow } from 'enzyme';
import React from 'react';
import configureStore, { MockStoreEnhanced } from 'redux-mock-store';
import { SideMenu } from './sideMenu';

describe('Side menu tests', () => {
  jest.mock(
    './../nav/sideNav',
    (): any => ({
      SideNav: (): string => 'SideNav'
    })
  );

  const initialState = {};
  const mockStore = configureStore();
  let store: MockStoreEnhanced;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders menu correctly', () => {
    const tree = shallow(<SideMenu />);
    expect(tree).toMatchSnapshot();
  });
});
