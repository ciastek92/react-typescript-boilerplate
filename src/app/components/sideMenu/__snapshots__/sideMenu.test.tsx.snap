// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side menu tests renders menu correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="o-aside-layout__burger u-display-md--none u-cursor--pointer js-hit-area"
          onClick={[Function]}
        >
          <Hamburger
            darkTheme={true}
            open={false}
          />
        </div>,
        <aside
          className="o-aside-layout__sidebar"
        >
          <SideNav>
            <div
              className="c-side-nav__burger u-cursor--pointer js-hit-area u-display--none"
              onClick={[Function]}
            >
              <Hamburger
                darkTheme={true}
                open={false}
              />
            </div>
          </SideNav>
        </aside>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Hamburger
            darkTheme={true}
            open={false}
          />,
          "className": "o-aside-layout__burger u-display-md--none u-cursor--pointer js-hit-area",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "darkTheme": true,
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SideNav>
            <div
              className="c-side-nav__burger u-cursor--pointer js-hit-area u-display--none"
              onClick={[Function]}
            >
              <Hamburger
                darkTheme={true}
                open={false}
              />
            </div>
          </SideNav>,
          "className": "o-aside-layout__sidebar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="c-side-nav__burger u-cursor--pointer js-hit-area u-display--none"
              onClick={[Function]}
            >
              <Hamburger
                darkTheme={true}
                open={false}
              />
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Hamburger
                darkTheme={true}
                open={false}
              />,
              "className": "c-side-nav__burger u-cursor--pointer js-hit-area u-display--none",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "darkTheme": true,
                "open": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        "type": "aside",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="o-aside-layout__burger u-display-md--none u-cursor--pointer js-hit-area"
            onClick={[Function]}
          >
            <Hamburger
              darkTheme={true}
              open={false}
            />
          </div>,
          <aside
            className="o-aside-layout__sidebar"
          >
            <SideNav>
              <div
                className="c-side-nav__burger u-cursor--pointer js-hit-area u-display--none"
                onClick={[Function]}
              >
                <Hamburger
                  darkTheme={true}
                  open={false}
                />
              </div>
            </SideNav>
          </aside>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Hamburger
              darkTheme={true}
              open={false}
            />,
            "className": "o-aside-layout__burger u-display-md--none u-cursor--pointer js-hit-area",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "darkTheme": true,
              "open": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SideNav>
              <div
                className="c-side-nav__burger u-cursor--pointer js-hit-area u-display--none"
                onClick={[Function]}
              >
                <Hamburger
                  darkTheme={true}
                  open={false}
                />
              </div>
            </SideNav>,
            "className": "o-aside-layout__sidebar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div
                className="c-side-nav__burger u-cursor--pointer js-hit-area u-display--none"
                onClick={[Function]}
              >
                <Hamburger
                  darkTheme={true}
                  open={false}
                />
              </div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Hamburger
                  darkTheme={true}
                  open={false}
                />,
                "className": "c-side-nav__burger u-cursor--pointer js-hit-area u-display--none",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "darkTheme": true,
                  "open": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": [Function],
          },
          "type": "aside",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
